### DISCUSSION TITLE
hi， imodel.js Do you support Boolean operations of mesh and solid?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#676] [Q&A] [answered]
hi， imodel.js Do you support Boolean operations of mesh and solid?I found that in regionops, there are operations involving 2D.Thanks.

**Comment by bbastings** [comment, marked as answer]
Can you describe the use case in more detail? If the goal is to create geometric elements, there is a new api in 2.11 that supports Boolean operations for creating the element's geometry. Be aware that it's still in the testing phase and marked as alpha.

Search for BRepGeometryOperation in:
https://github.com/imodeljs/imodeljs/blob/master/core/common/src/geometry/ElementGeometry.ts

For an example, search for "unite/subtract/intersect solids test" in:
https://github.com/imodeljs/imodeljs/blob/master/core/backend/src/test/standalone/GeometryStream.test.ts

**Comment by shenshengyi** [comment]
I want to perform Boolean operation between two solid or two mesh. Do  support it now? thanks。

**Reply by bbastings** [reply]
Yes, it's supported for creating elements as demonstrated in the example that was previously cited.
